### PR TITLE
Added solution hw-5.1 jv-clonable

### PR DESCRIPTION
**Pull request by nevskyy**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+public class Car implements Cloneable {
+    private String color;
+    private String body;
+    private String model;
+    private String regNumber;
+    private Engine engine;
 
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return new Car();
```
Comment (VitaliiLeshchenko) [inline src/main/java/core/basesyntax/Car.java:12]: this function don't do that what need

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+public class Engine implements Cloneable {
+    private String fuelSupplySystem;
+    private int cylinders;
+    private String horsePwr;
+    private String mark;
+    private int weight;
 
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return new Engine();
```
Comment (VitaliiLeshchenko) [inline src/main/java/core/basesyntax/Engine.java:12]: same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (nevskyy) [reply src/main/java/core/basesyntax/Car.java:12]: Thanks for comment well noted

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+        this.engine = engine;
+        this.color = color;
+        this.shape = shape;
+        this.model = model;
+        this.year = year;
+    }
+
+    @Override
+    public Car clone() {
+        return new Car(engine.clone(), color, model, year);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:20]: clone should copy all fields

Comment (nevskyy) [reply src/main/java/core/basesyntax/Car.java]: Pasha, good Morning. Missing field "shape" added as advised.

**Pull request closed without merging**